handwritten_support/riscv_extras_sequential.lem

```diff
@@ -153,6 +153,10 @@ val sys_enable_next : unit -> bool
 let sys_enable_next () = true
 declare ocaml target_rep function sys_enable_next = `Platform.enable_next`
 
+val sys_enable_fiom : unit -> bool
```

Comment:
This is hardcoded to true in the platform implementations, does it make any sense to add it there vs hardcoding it in sail?

Reply:
It just makes it configurable without changing the Sail code. Though you're right, I will add an emulator flag.

Maybe with a warning because it's supposed to be enabled if you support virtual memory and IIRC you can't disable that currently.

Reply:
Depends what privileged spec version you implement. We need to support all versions that people are making implementations for, so can't just go with the requirements of the latest version.

Reply:
Good point - that's something I've been wondering about actually. What version of the spec does the model target?

If it is "all of them" I would have expected some kind of `has_spec(Priv1_12)` checks but those don't seem to exist. For example the very first thing I checked from the privileged spec changes:

> Changed MRET and SRET to clear mstatus.MPRV when leaving M-mode.

is just done unconditionally in the model:

      if   cur_privilege != Machine
      then mstatus->MPRV() = 0b0;

Another example: the 1.12 spec also introduces `mstatush` which is unconditionally implemented in the model.

Perhaps the intention is to add something `has_spec(Priv1_12)` but it just hasn't been done yet? Or maybe the plan for supporting multiple spec versions hasn't been fully formed... Any insight appreciated!

Anyway, I'll just add an emulator flag for now. :-)

Reply:
Flag added.